### DISCUSSION TITLE
5 min intervals ? Is it hidden somewhere ?

### DISCUSSION THREAD

**Discussion by craigcurtin-dev** [amberelectric/public-api#38] [Q&A]
Myles,

WIth the publicly available API - i used to access the 5 min spot prices through Node Red - i have been using these to model the cost effectiveness of a battery system and feeding back into the grid.

We are about to pull the pin on getting our battery solution but i note the 5 minute prices seem to have disappeared (although the examples on the developer page do seem to show them when getting the current price) - i have tried to change the interval to 5 rather than 30 - but that seems to be ignored at this point.

I have dug through the full dailylisting and it does not appear to be in there either - everything is 30min intervals ?

Craig

**Comment by madpilot** [comment]
Correct - the API only supports 30-minute intervals at the moment, and we won't support true 5-minute intervals until 5-minute dispatch comes into play

I assume when talking about the old API, you mean intervals that had a source of 5MIN? Those aren't the current 5-minute spot price, they are the average of all the 5-minute actuals and the 30-minute forecast weighted by the network demand for those periods. 

The new API does the same thing, except it takes the 5-minute forecast into consideration. This is why the price changes every 5-minutes, even though we bill in 30-minute increments.

So, it's the same thing as you had before, except it responds a little bit quicker.

**Reply by craigcurtin-dev** [reply]
Myles,

Thanks for the quick response.

In my old scripts i used to run every 5 minutes (starting at 1 minute past the hour etc) i would extract all the prices and then search for the 5MIN - these would be updated every 5 minutes for the current interval - so i could watch the price spike that was forecast then analyse every 5 minutes to see how close the end price for the 1/2 hour was 

As an example if the Forecast prior to the start of a 1/2 hour period was a price of $3.00 then i would input to my battery control system to produce the maximum output, i would then have it continue to export at the maximum whilst the 5 minute intervals remained above $1.00, i would continue watching and if the price started to dip dramatically then i would start to back off the output 

Are you saying the new API in the currentInterval will be doing this averaging for me ? So if i call the API every minute i will see a rolling average for that 1/2 hour period ? - If so i assume there would be a dead spot in the last 5 minutes whilst the 1/2 hour price was finalised by NEM ?

Craig

**Reply by madpilot** [reply]
> Are you saying the new API in the currentInterval will be doing this averaging for me ? 

Bingo, though the old one did it for you too, it was just badly named. The range object might be useful to you too. If the range is big, the prices are volatile - if the range is small, you can be more certain of them (As certain as you can be when talking about predictions)

> So if i call the API every minute i will see a rolling average for that 1/2 hour period ? - If so i assume there would be a dead spot in the last 5 minutes whilst the 1/2 hour price was finalised by NEM ?

Yep, call it at around 45 seconds into the 5 minute period and it'll change (That's the earliest AEMO publishes the price data for that 5-minute period).

There is no dead spot the price you see in the last 5 minutes is the price you pay for the 30 minute period.

**Reply by craigcurtin-dev** [reply]
Thanks Myles,

Thats interesting with the old API (being badly named) as i thought it was just the bid price for the 5 minute interval each time rather than a running average - i will need to go back and have a look at that and the 1/2 hour prices to make sure my battery prediction Algo is correct.

WIll start playing around with the new API now on the current prices and see how that pricing looks

Thanks for all the help on this 

Craig

**Reply by craigcurtin-dev** [reply]
Hey,

I just wanted to open this one again as the data i am seeing now does not match what we discussed above.

With the price spikes in Sydney this evening - i am seeing on dashboard a different price to the one being returned by the API.

In my flow i am calling the API for a single current price at 1 minute past the hour and then every 5 minutes after that. Based on the discussion above - i should be seeing the current Averaged price being returned with each call - so by displaying them on my home dashboard i can then see how the price evolves from the start of the 1/2 hour block.

But tonight - where the price is within a big range during the peak - i am not seeing the same numbers - I would have thought that on YOUR dashboard - when you display a range in the NOW period the bottom of that range would be the current average price ? But thats not matching up with what the API is returning 

Am i approaching this the wrong way - or is there something whacky in your code when you display a range of prices ?

Craig

**Reply by madpilot** [reply]
We don't show the current price at all when we show a range. The range price you see in the API isn't adjusted for your network price - it's the range of the spot price.

This is a bug - it should match the app. (It's also ex GST, but that is another story)

**Reply by craigcurtin-dev** [reply]
Thanks for the quick response.

May be talking about two things here

When i am logged in to the website and display the Live Price on my PC - i am seeing a range for the price - currently showing 40c to $2.30

When i call the API for the current record only and access the PriceKwh in there - i would expect that this would return the rolling averaged value for each 5 minute interval (as discussed above). i would also think (logically) that this averaged price would be the bottom of the price range on my PC - but it does not seem to be ?

Craig

**Reply by madpilot** [reply]
So `priceKwh` is the rolling average. That will be in the API, but you won't see it in the web app or mobile app if a range is displayed.

The `min` and `max` value are influenced by but aren't the same as the weighted average. It's the range of the spot price, not `priceKWH`. And it doesn't include GST. So your assumption isn't right - min rarely equals the current price.

Does that explain it?

**Reply by madpilot** [reply]
In the app - the range is based around the `perKWH` price, which is why it looks super wrong

**Reply by craigcurtin-dev** [reply]
Aaah - that does explain it - my assumption was that Min would be the current rolling avg of priceKwh - you are however saying their is some other magic sauce in the calcs for Min and Max

I understand what you are saying - but not sure the reasoning behind it - but at least i know which i can trust at this point.

Craig

**Reply by madpilot** [reply]
The point of the min and max number are to show volatility. It's confusing and we're reviewing at the moment.

But from an API point of view it can still be interesting.

AEMO published the results of their 100 or so models that they use to predict the price. We effectively take the max and min of those for our ranges. The current price range is a little more complicated as it takes previous and next max values in to account too, but ultimately it's a good indicator that the AEMO models are disagreeing so expect the price to be a bit all over the shop.

**Reply by tjhowse** [reply]
So am I correct in thinking that starting 2021-10-01 we'll see `'duration': 5` interval responses to `prices/current` calls? If so, exciting times!

**Reply by madpilot** [reply]
Not unless you specifically request it. If you omit the resolution query param will still get 30-minute data back.

**Reply by madpilot** [reply]
Not sure if there is weird threading going on, but I'll post this explanation of the current state of play for 5-minute dispatch just in case: https://github.com/amberelectric/public-api/discussions/65#discussioncomment-1364794
